cheeseshop/dbapi.py

```diff
@@ -311,10 +311,79 @@ class CsGoSteamId(object):
         await conn.execute('''
             CREATE TABLE cs_go_steam_ids(
                 id serial PRIMARY KEY,
-                steam_id text UNIQUE NOT NULL
+                steam_id text UNIQUE NOT NULL,
+                print_name text,
+                team integer REFERENCES cs_go_team_ids (id)
             )
         ''')
 
+    @staticmethod
+    async def get_all(conn):
+        players = []
+        async for record in conn.cursor('''
+            SELECT * from cs_go_steam_ids
+        '''):
+            players.append({
+                'id': record['steam_id'],
+                'name': record['print_name'],
+                'team': record['team_name}']})
+        return players
+
+
+class CsGoTeamNames(object):
+    @staticmethod
+    async def create_schema(conn):
+        await conn.execute('''
+            CREATE TABLE cs_go_team_names(
+                id serial PRIMARY KEY,
+                team integer REFERENCES cs_go_team_ids (id),
+                team_name text
+            )
+        ''')
+
+    @staticmethod
+    async def get_names_by_id(conn, team_id):
+        names = []
+        async for record in conn.cursor('''
+            SELECT * FROM cs_go_team_names
+            WHERE id = $1
+        ''', team_id):
+            names.append(record['team_name'])
+        return names
+
+    @staticmethod
+    async def get_ids_by_name(conn, team_name):
+        ids = []
+        async for record in conn.cursor('''
+            SELECT * FROM cs_go_team_names
+            WHERE team_name LIKE %$1$%
+        ''', team_name):
+            ids.append(record['team'])
+        return ids
+
+    def __init__(self, id_, name):
+        self.id = id
+        self.name = name
+
+
+class CsGoTeam(object):
+    @staticmethod
+    async def create_schema(conn):
+        await conn.execute('''
+            CREATE TABLE cs_go_team_ids(
+                id serial PRIMARY KEY
```

Comment:
This table is going to require some thinking. If we have no information to distinguish a 'team' other than name then wouldnt two differently named teams always be considered different teams (and therefore name for a team would belong in this table)? I suspect the idea is we want to use some other information to correlate events from a team even if the name isnt exatly the same. If so, then we have to figure out what other information we are going to use and that information has to be added as a column in this table.

Reply:
Sounds like theres an hltv id which is the main source of truth we should try for disambiguating a team. Lets add a hltv_id column here